controllers/openstackdataplanenode_controller.go

```diff
@@ -233,40 +233,47 @@ func (r *OpenStackDataPlaneNodeReconciler) Reconcile(ctx context.Context, req ct
-		result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
-			ctx,
-			helper,
-			instance,
-			instance,
-			ansibleSSHPrivateKeySecret,
-			nodeConfigMap,
-			&instance.Status,
-			instance.Spec.NetworkAttachments,
-			instance.Spec.OpenStackAnsibleEERunnerImage)
-		if err != nil {
-			return result, err
-		}
+		if instance.Spec.Node.Nova != nil {
+			var novaExternalCompute *novav1beta1.NovaExternalCompute
+			result, novaExternalCompute, err = deployment.DeployNovaExternalCompute(
+				ctx,
+				helper,
+				instance,
+				instance,
+				ansibleSSHPrivateKeySecret,
+				nodeConfigMap,
+				&instance.Status,
+				instance.Spec.NetworkAttachments,
+				instance.Spec.OpenStackAnsibleEERunnerImage,
+				// There is anther nova template accessible in
+				// instanceRole.Spec.NodeTemplate.Nova
+				// Which one we should use here? Inheritance?
+				instance.Spec.Node.Nova,
+			)
+			if err != nil {
+				return result, err
+			}
 
-		novaReadyCondition := novaExternalCompute.Status.Conditions.Get(condition.ReadyCondition)
-		if novaReadyCondition != nil {
-			r.Log.Info(fmt.Sprintf("NovaExternalCompute ReadyCondition status: %s", novaReadyCondition.Status))
-		} else {
-			r.Log.Info("NovaExternalCompute ReadyCondition not yet set")
-		}
+			novaReadyCondition := novaExternalCompute.Status.Conditions.Get(condition.ReadyCondition)
+			if novaReadyCondition != nil {
+				r.Log.Info(fmt.Sprintf("NovaExternalCompute ReadyCondition status: %s", novaReadyCondition.Status))
+			} else {
+				r.Log.Info("NovaExternalCompute ReadyCondition not yet set")
+			}
 
-		mirroredCondition := novaExternalCompute.Status.Conditions.Mirror(dataplanev1beta1.NovaComputeReadyCondition)
-		if mirroredCondition != nil {
-			instance.Status.Conditions.Set(mirroredCondition)
-		}
+			mirroredCondition := novaExternalCompute.Status.Conditions.Mirror(dataplanev1beta1.NovaComputeReadyCondition)
+			if mirroredCondition != nil {
+				instance.Status.Conditions.Set(mirroredCondition)
+			}
 
-		if condition.IsError(instance.Status.Conditions.Get(dataplanev1beta1.NovaComputeReadyCondition)) {
-			r.Log.Info(fmt.Sprintf("%s error", dataplanev1beta1.NovaComputeReadyCondition))
-			err = fmt.Errorf("failed: NovaExternalCompute name %s NovaExternalCompute namespace %s", novaExternalCompute.Name, novaExternalCompute.Namespace)
-			return ctrl.Result{}, err
+			if condition.IsError(instance.Status.Conditions.Get(dataplanev1beta1.NovaComputeReadyCondition)) {
+				r.Log.Info(fmt.Sprintf("%s error", dataplanev1beta1.NovaComputeReadyCondition))
+				err = fmt.Errorf("failed: NovaExternalCompute name %s NovaExternalCompute namespace %s", novaExternalCompute.Name, novaExternalCompute.Namespace)
+				return ctrl.Result{}, err
+			}
 		}
 
-		if instance.Status.Conditions.IsTrue(dataplanev1beta1.NovaComputeReadyCondition) {
+		if instance.Spec.Node.Nova == nil || instance.Status.Conditions.IsTrue(dataplanev1beta1.NovaComputeReadyCondition) {
```

Comment:
maybe it worth noting that other Spec.Node.Foo as get defined, should all be added to this condition